larray/core/group.py

```diff
@@ -911,20 +915,31 @@ def by(self, length, step=None):
         >>> from larray import Axis, X
         >>> age = Axis(range(10), 'age')
         >>> age[[1, 2, 3, 4, 5]].by(2)
-        (age[1, 2], age[3, 4], age[5])
+        (age[1, 2] >> '1:2', age[3, 4] >> '3:4', age[5] >> '5')
         >>> age[1:5].by(2)
-        (age.i[1:3], age.i[3:5], age.i[5:6])
+        (age.i[1:3] >> '1:2', age.i[3:5] >> '3:4', age.i[5:6] >> '5')
         >>> age[1:5].by(2, 4)
-        (age.i[1:3], age.i[5:6])
+        (age.i[1:3] >> '1:2', age.i[5:6] >> '5')
         >>> age[1:5].by(3, 2)
-        (age.i[1:4], age.i[3:6], age.i[5:6])
+        (age.i[1:4] >> '1:3', age.i[3:6] >> '3:5', age.i[5:6] >> '5')
+        >>> age[1:5].by(3, 2, '{start}->{end}')
+        (age.i[1:4] >> '1->3', age.i[3:6] >> '3->5', age.i[5:6] >> '5')
         >>> X.age[[0, 1, 2, 3, 4]].by(2)
-        (X.age[0, 1], X.age[2, 3], X.age[4])
+        (X.age[0, 1] >> '0:1', X.age[2, 3] >> '2:3', X.age[4] >> '4')
         """
+        def make_group(start, length, name_template):
+            g = self[start:start + length]
+            labels = g.eval()
+            if labels[0] != labels[-1]:
+                g.name = name_template.format(start=labels[0], end=labels[-1])
+            else:
+                g.name = str(labels[0])
+            return g
         if step is None:
             step = length
-        return tuple(self[start:start + length]
-                     for start in range(0, len(self), step))
+        if name_template is None:
+            name_template = '{start}:{end}'
+        return tuple(make_group(start, length, name_template) for start in range(0, len(self), step))
```

Comment:
I thought #669 was just about having a make_group do something like:

```python
        def make_group(start, length):
            g = self[start:start + length]
            g.name = str(g)
            return g
```
but handling single label groups like you did is probably a good idea.